src/server/modes/charts/plugins/datalens/preparers/line/helpers/line-mappers/prepareLines.ts

```diff
@@ -345,8 +345,12 @@ export const prepareLines = (args: PrepareLinesArgs) => {
         const currentLine = lines[keys.key];
 
         if (segmentNameKey) {
+            // old idea:
             // segments should not use unique id, because id is used for legend item click in chartkit
             // it leads to problem with hiding and displaying data.
+            // update:
+            // anyway, we need to set id (it is set below), so filtering by legend will work as expected
+            // see CHARTS-10035
```

Comment:
I suggest removing comments about old/new ideas - they can be included in the description of pr (and then found if it is really necessary)